src/evict/evict_lru.c

```diff
@@ -1957,11 +1965,11 @@ __evict_walk_tree(WT_SESSION_IMPL *session, WT_EVICT_QUEUE *queue, u_int max_ent
         /* Each time when evict_ref is null, alternate between linear and random walk */
```

Comment:
It's unclear to me, but should the `WT_READ_PREV` flag be cleared in the `WT_EVICT_WALK_NEXT` case?

Reply:
```
*walk_flagsp = WT_READ_CACHE | WT_READ_NO_EVICT | WT_READ_NO_GEN | WT_READ_NO_WAIT;
```

We haven't set the flag in the previous code. Why we need to clear it here?

Reply:
The field `WT_READ_PREV` is set in the `_PREV` cases, but I don't see it cleared. Perhaps this implicitly or explicitly happens at a higher layer?

Reply:
Every time we use a new walk flags. If we have set the flag WT_READ_PREV in the previous tree walk, it is discarded after that.

Reply:
Okay got it thanks.